File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int max_h = 0;
    int max_w = 0;
  
    for(int i=0; i<sizes.size(); i++) {
        if(sizes[i][0] > sizes[i][1]) swap(sizes[i][0],sizes[i][1]);
    }
    for(int i=0; i<sizes.size(); i++) {
        max_w = max(max_w,sizes[i][1]);
    }
    for(int i=0; i<sizes.size(); i++) {
        max_h = max(max_h,sizes[i][0]);
    }
    answer = max_w * max_h;
    
    
    return answer;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    var answer = 1;
    let arr1=[];
    let arr2=[];
    for(size of sizes) {
        if(size[0] > size[1]) {
            arr2.push(size[0]);
            arr1.push(size[1]);
        }
        else {
            arr1.push(size[0]);
            arr2.push(size[1]);
            
        }
    }
    answer *= Math.max(...arr1);
    answer *= Math.max(...arr2);
    return answer;
}

